keps/349-avoid-coscheduling/README.md

```diff
@@ -0,0 +1,350 @@
+      - [Integration tests](#integration-tests)
+    - [Graduation Criteria](#graduation-criteria)
+  - [Implementation History](#implementation-history)
+  - [Drawbacks](#drawbacks)
+  - [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This proposal introduces a mechanism for avoid co-scheduling of workload pods.
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+Some workloads need all pods to be running at the same time to make progress, for
+example, when they require pod-to-pod communication. In that case a large
+workload may have difficulty scheduling all of its pods for a prolonged time as
+smaller workloads would continue to block the slots. Moreover, a pair of
+workloads may deadlock if the amount of actual resources available is below the
+quota configured to determine admission.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- a mechanism to avoid concurrent scheduling of pods for workloads.
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+- guarantee that two workloads would not schedule pods concurrently. Example
+scenarios in which two workloads may still concurrently schedule their pods:
+  - when succeeded pods are replaced with new because job's parallelism is less than its completions;
+  - when a failed pod gets replaced.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+We introduce a mechanism to mark a Job as blocking admission of further
+workloads until the first batch of its pods is scheduled, more precisely, until:
+`job.status.ready + job.status.succeeded >= job.spec.parallelism`.
```

Comment:
Also say that this is an option for the workload

Reply:
Added a note